### PR TITLE
More scripts

### PR DESCRIPTION
**Pull request by fisx**

### CODE REVIEW

**Comment by fisx** [comment]
I am going to tests these later (unless somebody will be faster than me), but this is ready for review.

File: deploy/services-demo/create_team_members.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        -H'Content-type: application/json' \
+        -H'Z-User: '"$ADMIN_UUID"'' \
+        -d'{"email":"'"$EMAIL"'","name":"'"$USER_NAME"'","inviter_name":"Team admin"}')
+
+    INVITATION_ID=$(echo "$CURL_OUT_INVITATION" | tail -1 | sed 's/.*\"id\":\"\([a-z0-9-]*\)\".*/\1/')
+
+    echo "Created the invitation, sleeping 1 second..." 1>&2
+    sleep 1
+
+    ERR='{"code":409,"message":"The given e-mail address is in use.","label":"email-exists"}'
```
Comment (jschaul) [inline deploy/services-demo/create_team_members.sh]: This is not the only error case; the team ID or the user ID could also be incorrect, or the permissions for that admin_userid could not be sufficient. Perhaps you could abort the script on any 4xx, not just this particular 409.

File: deploy/services-demo/create_team_members.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+
+set -ex
```
Comment (jschaul) [inline deploy/services-demo/create_team_members.sh]: is `-x` verbosity necessary?

File: deploy/services-demo/create_team_members.sh [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+while getopts ":a:t:h:c" opt; do
+  case ${opt} in
+    a ) ADMIN_UUID="$OPTARG"
+      ;;
+    t ) TEAM_UUID="$OPTARG"
+      ;;
+    h ) BRIG_HOST="$OPTARG"
+      ;;
+    c ) if [ ! -e "$OPTARG" ]; then
+            echo -e "\n\n*** I need the name of an existing cvs file.\n\n"
```
Comment (jschaul) [inline deploy/services-demo/create_team_members.sh]:
```suggestion
            echo -e "\n\n*** I need the name of an existing csv file.\n\n"
```

File: deploy/services-demo/create_team_members.sh (hunk 2 above)
Comment (fisx) [reply deploy/services-demo/create_team_members.sh]: I like it.  If you don't let me know and I'll remove it.

File: deploy/services-demo/create_team_members.sh (hunk 1 above)
Comment (fisx) [reply deploy/services-demo/create_team_members.sh]: like this?: https://github.com/wireapp/wire-server/pull/801/commits/311c13c902db3cd209ed98d616faebd1413aa1c2

**Comment by fisx** [comment]
tried this on staging:

```
./create_team_members.sh -a ... -t ... -h http://localhost:8080 -c test.csv
```

with

```
13d36c88-a7e6-11e9-b223-17f9008c907c@mailinator.com,Frau Eins
191979ee-a7e6-11e9-95dd-33b87ece1d4e@mailinator.com,Herr Zwei
1a2e785c-a7e6-11e9-b369-db410f934c12@mailinator.com,Grünling
```

2019-07-16-182703_2048x1152_scrot

mailinator doesn't work any more without registration, but seeing pending users is good enough for me.

File: deploy/services-demo/create_team_members.sh [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+is a csv file with email addresses and suggested user names.
+
+Note that this uses internal brig endpoints.  It is not exposed over
+nginz and can only be used if you have direct access to brig.
+
+USAGE: $0
+    -a <admin uuid>: User ID of the inviting admin.  default: ${ADMIN_UUID}
+    -t <team uuid>: ID of the inviting team.  default: ${TEAM_UUID}
+    -h <host>: Base URI of brig. default: ${BRIG_HOST}
+    -c: input file containing info on the invitees in format 'Email,UserName'.  default: ${CSV_FILE}
```
Comment (julialongtin) [inline deploy/services-demo/create_team_members.sh]: perhaps "-c <input_file>: file containing ...

File: deploy/services-demo/create_test_team_members.sh [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+if [ "$#" -ne 0 ]; then
+  echo "$USAGE" 1>&2
+  exit 1
+fi
+
+# Generate users
+END=$((COUNT + START - 1))
+for i in $(seq "$START" "$END")
+do
+    EMAIL='w'$(printf "%03d" "$i")"@example.com"
```
Comment (julialongtin) [inline deploy/services-demo/create_test_team_members.sh]: perhaps a comment or something to tell the user the emails start with 'w'?

File: deploy/services-demo/create_test_team_members.sh [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    CURL_OUT=$(curl -i -s --show-error \
+            -XPOST "$BRIG_HOST/i/users" \
+            -H'Content-type: application/json' \
+            -d'{"email":"'"$EMAIL"'","password":"'"$PASSWORD"'","name":"demo","team_code":"'"$INVITATION_CODE"'"}')
+
+    TEAM_MEMBER_UUID=$(echo "$CURL_OUT" | tail -1 | sed 's/.*\"id\":\"\([a-z0-9-]*\)\".*/\1/')
+    TEAM=$(echo "$CURL_OUT" | tail -1 | sed 's/.*\"team\":\"\([a-z0-9-]*\)\".*/\1/')
+
+    if [ "$TEAM" != "$TEAM_UUID" ]; then
+        echo "unexpected error: user got assigned to no / the wrong team?!"
```
Comment (julialongtin) [inline deploy/services-demo/create_test_team_members.sh]: perhaps dump the curl_out here?

File: deploy/services-demo/create_test_team_members.sh (hunk 5 above)
Comment (tiago-loureiro) [reply deploy/services-demo/create_test_team_members.sh]: One thing that we should be careful about is that we _actually_ send out emails with this request.
Sending emails to _example.com_ will trigger an email being sent out and may lower our reputation.
Perhaps make the domain configurable and explain that, currently, we _effectively_ send out the emails (and perhaps we need an extra endpoint/flag that prevents sending it over the Wire?)
@fisx

Comment (fisx) [reply deploy/services-demo/create_test_team_members.sh]: Sorry, only saw that after I merged...  I'll make another PR and ping you.

**Pull request merged**